internal/pkg/epinject/epinject.go

```diff
@@ -195,11 +195,5 @@ func withConnectionHelper(c *client.Client) error {
 }
 
 func containsEntrypoint(entrypoint []string) bool {
-	for _, ep := range entrypoint {
-		if ep == "/waypoint-entrypoint" {
-			return true
-		}
-	}
-
-	return false
+	return entrypoint[0] == "/waypoint-entrypoint"
```

Comment:
You probably have to check that entrypoint is at least 1 element to avoid a panic.